icicle/appUtils/ntt/ntt.cuh

```diff
@@ -321,10 +360,13 @@ __global__ void ntt_template_kernel(E *arr, uint32_t n, S *twiddles, uint32_t n_
       uint32_t k = i + j + shift_s;
 
       uint32_t offset = (task / chunks) * n;
-      E u = arr[offset + i + j];
-      E v = twiddles[j * n_twiddles_div] * arr[offset + k];
-      arr[offset + i + j] = u + v;
-      arr[offset + k] = u - v;
+      if (!rev)
```

Comment:
pls review and try to use a local variable if possible - it like 2 read + 2 write (not counting twiddles) instead of 4 + 6 (including conditional) to global memory

Reply:
You're right, the old version is slightly faster, checked with criterion benchmarks for NTT (will include them in the next commit). Thought that removing 1 local variable would be worth it, but it's apparently outweighed by extra global memory accesses